pandas/core/arrays/integer.py

```diff
@@ -0,0 +1,563 @@
+    def _create_comparison_method(cls, op):
+        def cmp_method(self, other):
+
+            op_name = op.__name__
+            mask = None
+            if isinstance(other, IntegerArray):
+                other, mask = other.data, other.mask
+            elif is_list_like(other):
+                other = np.asarray(other)
+                if other.ndim > 0 and len(self) != len(other):
+                    raise ValueError('Lengths must match to compare')
+
+            # numpy will show a DeprecationWarning on invalid elementwise
+            # comparisons, this will raise in the future
+            with warnings.catch_warnings(record=True):
+                with np.errstate(all='ignore'):
+                    result = op(self.data, other)
+
+            # nans propagate
+            if mask is None:
+                mask = self.mask
+            else:
+                mask = self.mask | mask
+
+            result[mask] = True if op_name == 'ne' else False
+            return result
+
+        name = '__{name}__'.format(name=op.__name__)
+        return set_function_name(cmp_method, name, cls)
+
+    def _maybe_mask_result(self, result, mask, other, op_name):
+        """
+        Parameters
+        ----------
+        result : array-like
+        mask : array-like bool
+        other : scalar or array-like
+        op_name : str
+        """
+
+        # may need to fill infs
+        # and mask wraparound
+        if is_float_dtype(result):
+            mask |= (result == np.inf) | (result == -np.inf)
+
+        # floor div can be a float or an integer dependending
+        # on the operands
+        if (op_name in ['rfloordiv', 'floordiv'] and
+                (is_float_dtype(other) or is_float(other))):
+            result[mask] = np.nan
+            return result
+
+        # by definition a float result
+        elif op_name in ['rtruediv', 'truediv', 'rdiv', 'div']:
+            result[mask] = np.nan
+            return result
+
+        elif is_float_dtype(result):
+            # if our float result, try to downcast if possible
+            # if remains float, then mask and return as float
+            nonans = result[notna(result)]
+            maybe = maybe_downcast_to_dtype(nonans, self.dtype.numpy_dtype)
+            if not is_integer_dtype(maybe):
+                result[mask] = np.nan
+                return result
+
+        return type(self)(result, mask=mask, dtype=self.dtype, copy=False)
+
+    @classmethod
+    def _create_arithmetic_method(cls, op):
+        def integer_arithmetic_method(self, other):
+
+            op_name = op.__name__
+            mask = None
+            if isinstance(other, (ABCSeries, ABCIndexClass)):
+                other = getattr(other, 'values', other)
+
+            if isinstance(other, IntegerArray):
+                other, mask = other.data, other.mask
+            elif getattr(other, 'ndim', 0) > 1:
+                raise TypeError("can only perform ops with 1-d structures")
```

Comment:
return `NotImplemented` for `DataFrame`?

Reply:
fixed